go/database/vt/state_test.go

```diff
@@ -0,0 +1,486 @@
+package vt
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/0xsoniclabs/carmen/go/backend"
+	"github.com/0xsoniclabs/carmen/go/common"
+	"github.com/0xsoniclabs/carmen/go/common/amount"
+	"github.com/0xsoniclabs/carmen/go/common/immutable"
+	"github.com/0xsoniclabs/carmen/go/common/witness"
+	"github.com/0xsoniclabs/carmen/go/database/vt/geth"
+	"github.com/0xsoniclabs/carmen/go/database/vt/memory"
+	"github.com/0xsoniclabs/carmen/go/state"
+	"github.com/stretchr/testify/require"
+)
+
+func TestState_CreateAccount_Hash_Matches(t *testing.T) {
+	state, err := initTestedStates(t)
+	require.NoError(t, err, "failed to initialize tested states")
+	defer func() {
+		require.NoError(t, state.Close(), "failed to close state")
+	}()
+
+	addr := common.Address{1}
+
+	update := common.Update{}
+	update.CreatedAccounts = append(update.CreatedAccounts, addr)
```

Comment:
I have been running a test removing the `CreateAccounts` field from the `Update` struct, and the Sonic block history could be successfully replayed. Also, when dumping and [restoring](https://github.com/0xsoniclabs/carmen/blob/5434be3ba2df3e74c557c129d503550a2aa6ed13/go/database/mpt/io/archive.go#L272-L363) archive state we use deltas between states and we do not require calls to the creation of an account.

I thus suspect that we do not really need the `CreateAccounts` field in the `Update` struct and we do not have to worry about its implementation. We should actually attempt to remove it.